elements/includes/creative_commons.inc

```diff
@@ -203,59 +182,505 @@ function xml_form_elements_creative_commons($element, &$form_state) {
+    $jurisdiction = $jurisdiction_default;
+    $license_uri = $license_uri_default;
+  }
+  if (!in_array($license, array('none', 'manual'))) {
+    // Check if there is a newer version unless user sets the URI manually.
+    $current_query = xml_form_elements_creative_commons_make_value_array($license, $derivatives, $commercial, $jurisdiction, '', '');
+    $response = FALSE;
+    if (isset($form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get']['query']) &&
+      $form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get']['query'] == $current_query &&
+      isset($form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get']['response']) &&
+      ($form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get']['response'] !== FALSE)) {
+      // Get response from storage.
+      $response = simplexml_load_string($form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get']['response'], 'SimpleXMLElement');
+    }
+    else {
+      $response = xml_form_elements_get_creative_commons($commercial, $derivatives, $jurisdiction, FALSE, '', $license);
+      // Cache response.
+      if (!isset($form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get'])) {
+        $form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get'] = array();
+      }
+      $form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get']['query'] = $current_query;
+      $form_state['storage']['xml_form_elements'][$element['#name']]['api_data_get']['response'] = ($response !== FALSE) ? $response->asXML() : FALSE;
+    }
+    if ($response) {
+      $license_uri = (string) $response->{'license-uri'};
+      $license_value_array = xml_form_elements_creative_commons_parse_uri($license_uri);
+      $new_version = $license_value_array[3];
+      if ($version != '') {
+        if ((float) $new_version < (float) $version) {
+          // The queried version is too high.
+          $error = TRUE;
+          form_set_error($element['#name'] . '][license_fieldset][license_version', t("CC license version not (yet) available (leave empty for newest)."));
+        }
+        elseif ((float) $new_version == (float) $version) {
+          // The queried version is already the newest.
+          $version = '';
+        }
+        else {
+          // The queried version is lower and we need to replace it in the URI.
+          $license_uri = str_replace($new_version, $version, $license_uri);
+          if ($license_uri == XML_FORM_ELEMENTS_CREATIVE_COMMONS_LICENSE_URI_DEFAULT) {
+            // Warn that the module default seems to be deprecated.
+            watchdog('xml_form_elements', 'Creative Commons default license seems to be deprecated. Please announce the developers.', array(), WATCHDOG_WARNING);
+          }
+          $current_query = xml_form_elements_creative_commons_make_value_array('', '', '', '', '', $license_uri);
```

Comment:
Sorry for Travis' hiccup. I will mend the "[creative_commons.inc#L422-464](https://github.com/d-r-p/islandora_xml_forms/blob/1e821c0fe7078207d6344aaa1bf1b101cf868510/elements/includes/creative_commons.inc#L442-L464) and [creative_commons.inc#L489-511](https://github.com/d-r-p/islandora_xml_forms/blob/1e821c0fe7078207d6344aaa1bf1b101cf868510/elements/includes/creative_commons.inc#L489-L511) are duplicates" as soon as I get to it...